[v16] Simplify the docs site navigation

ptgott opened the pull request:
Backports #44327
Backports part of #45175
Backports #45229

Provide a less overwhelming documentation experience for users by condensing sidebar sections into five, organized loosely around different audiences of the documentation:

- **Get Started:** new Teleport users, as well as users of all levels who want to quickly look up key, frequently changing information, such as the changelog and upcoming release information.
- **Admin Guides:** intermediate users completing a specific task. This includes all how-to guides in the docs except those in "Get Started" and "Enroll Resources".
- **Enroll Resources:** Admin Guides, but for our extensive library of how-to guides for protecting infrastructure resources with Teleport.
- **References:** Advanced users looking for comprehensive reference information and architecture guides. This includes all reference guides in the docs.
- **User Guides:** Guides for end users looking to access resources.

Related changes:
- **Move Helm reference partials**. Use the `docs/pages/includes` directory so we can generate the sidebar from `docs/pages/reference`.
- **Move the Architecture sidebar section** into "References", and move architecture guides across the documentation into "Architecture".
- **Remove "Documentation Overview"** as this is out of date. Now, top-level sidebar sections are more self explanatory.
- **Remove "Choose an Edition"**, since we want to elevate Teleport Enterprise (Cloud), making Teleport Enterprise (Self-Hosted) available for certain use cases.
- **Add an infrastructure as code section** to Admin Guides.

ptgott committed:
Simplify the docs site navigation

Backports #44327

Provide a less overwhelming documentation experience for users by
condensing sidebar sections into five, organized loosely around
different audiences of the documentation:

- **Get Started:** new Teleport users, as well as users of all levels
  who want to quickly look up key, frequently changing information, such
  as the changelog and upcoming release information.
- **Admin Guides:** intermediate users completing a specific task. This
  includes all how-to guides in the docs except those in "Get Started"
  and "Enroll Resources".
- **Enroll Resources:** Admin Guides, but for our extensive library of
  how-to guides for protecting infrastructure resources with Teleport.
- **References:** Advanced users looking for comprehensive reference
  information and architecture guides. This includes all reference
  guides in the docs.
- **User Guides:** Guides for end users looking to access resources.

Related changes:
- **Move Helm reference partials**. Use the `docs/pages/includes`
  directory so we can generate the sidebar from `docs/pages/reference`.
- **Move the Architecture sidebar section** into "References", and move
  architecture guides across the documentation into "Architecture".
- **Remove "Documentation Overview"** as this is out of date. Now, top-level
  sidebar sections are more self explanatory.
- **Remove "Choose an Edition"**, since we want to elevate Teleport
  Enterprise (Cloud), making Teleport Enterprise (Self-Hosted) available
  for certain use cases.
- **Add an infrastructure as code section** to Admin Guides.

hugoShaka committed:
Fix terraform provider reference index (#45175)

hugoShaka commented:
The "login rule via Terraform" slug looks like `/admin-guides/infrastructure-as-code/terraform/` ([See the preview docs](https://goteleport.com/docs/ver/17.x/admin-guides/infrastructure-as-code/terraform/)). Can we fix this before it lands on live v16 documentation? Else we'll need to do a redirect and coin the `terraform` name.

ptgott commented:
@hugoShaka Thanks for catching this! I'll get a fix merged before we merge the reorganization into v16

hugoShaka commented:
@ptgott could you also cherry-pick b42c0558bb0b3e4624602c23c6d561b336074b57 from https://github.com/gravitational/teleport/pull/45198 before merging 🙏 ?

ptgott committed:
Reorganize Teleport Policy docs

Backports #45229

The Teleport Policy docs are currently in two sections of
`/admin-guides/access-controls` with overlapping subject areas. This
change merges the two sections into a single subsection of
`/admin-guides`.

To preserve the convention of moving all self-hosted guides into
`/admin-guides/deploy-a-cluster`, this change also moves guides to
self-hosting the Access Graph Service into
`/admin-guides/deploy-a-cluster/access-graph`.

More specific changes:

- Add a subsection of the Policy section for integrations.
- Make the Policy section a top-level section within Admin Guides, since
  we add another directory, to keep the max sidebar depth at four.
- Move the contents of `/admin-guides/access-controls/access-graph` to
  the `/admin-guides/teleport-policy/integrations` directory, since all
  of the guides in `access-graph` had to do with integrations.
- Change the Teleport Policy Integrations page into a menu. The page
  overlaps with the instructions on the AWS integration, so repurpose
  the guide as an overview.
- Turn the Policy Get Started guide into a menu. The guide is a brief
  overview, so it functions well as the introduction to a menu page.
- Rename the usage guide so it appears first on the auto-generated
  sidebar section.

ptgott commented:
> The "login rule via Terraform" slug looks like `/admin-guides/infrastructure-as-code/terraform/` ([See the preview docs](https://goteleport.com/docs/ver/17.x/admin-guides/infrastructure-as-code/terraform/)). Can we fix this before it lands on live v16 documentation? Else we'll need to do a redirect and coin the `terraform` name.
> 
> Edit: same issue for the "login rule via the Operator" page, its slug is `/admin-guides/infrastructure-as-code/kubernetes/`

@hugoShaka I have a PR to fix this here: https://github.com/gravitational/teleport/pull/45269

ptgott commented:
@zmb3 Checking to see if you have time to give this backport a look. Thanks!

fheinecke commented on CHANGELOG.md:
Are we certain that this is a desirable change (even if it breaks paths)? This will no longer match the changelog in already released artifacts.

fheinecke reviewed (commented):
Can you run `make update-version` on another branch, prior to merge, to confirm that releases won't break?

fheinecke commented:
> If this is going to break integrity checks, though, should we render the CHANGELOG on the docs site from master instead?

This shouldn't break any actual check we have - just didn't know if this behavior was desirable. I think we're fine changing old changelog entries.
